runner/execer/os/execer.go

```diff
@@ -197,11 +197,29 @@ func (e *execer) monitorMem(p *process, memCh chan scootexecer.ProcessStatus) {
 					Error:    msg,
 					ExitCode: 1,
 				}
+				// check whether memory consumption is above threshold immediately on process start
+				// mostly indicates that memory utilization was already above cap when the process started
+				if thresholdsIdx == 0 {
```

Comment:
this is sort of confusing... are we checking if thresholdsIdx is 0 because that means we haven't measured the memory above 25% of the cap yet?
I think this would be easier to understand if we refactored this monitorMem func a bit. Pull out the block inside of `if mem >= e.memCap`, e.g.`if mem >= e.memCap {
    memCapKillHandler()
}`, and then we could just explicitly check the mem usage once before we start the monitorMem goroutine

Reply:
We're checking if threshold index is 0 because it implies that it's the first time we're checking memory. I agree that this could be a bit confusing, going to refactor this.

Reply:
Refactored